Initial version of the Readme page in English (#121)
tlbootcamp/tlroadmap: translations/README-en.md

![Logotype](../logo.png)

> Team Lead is a ❄️, because in each company they are unique and distinct.

- [Roadmap](#roadmap)

> - [Telegram chat TL Bootcamp (RU)](https://tlinks.run/tlbootcamp)

Reviewer: Давай спилим вообще, наверное. Если пойдет в массы, сделаем английскую версию.

# What is it about?

> The concepts of "Team Lead", "Tech Lead", "Tech Manager" and others are very stretchy. Companies solve different tasks, these roles have different responsibilities. Besides, different leadership styles, differences in personal skills, and much more are put on the picture. But isn't that what happens to the developers? Different companies, different tasks, again, different nature of tasks and responsibilities.

Reviewer 2: ```suggestion
The concepts of "Team Lead", "Tech Lead", "Tech Manager" and others are very stretchy. Companies solve different tasks, these roles have different responsibilities. Besides, various leadership styles, differences in personal skills, and much more are put on the picture. But isn't that what happens to the developers? Different companies, different tasks, again, different nature of tasks and responsibilities.
```
?

We interviewed several dozen large and small companies, found out what they expect from Team Leads and how they interact with them. As a result, we were able to form a single model of basic teamwork competencies that covers all the cases studied.

> This model can be used in any way you like - to come up with your development plan, to create job descriptions in companies, to write a job description or to conduct interviews. Keep in mind it is likely you don't need all the branches of potential development - and this is fine.

Reviewer: В оригинале "должностные инструкции"/"описания вакансий". Здесь одно и то же выражение "job description", выглядит странно.

Reviewer 3: "define duties and responsibilities"

> > 🚨 **Important:** Depending on the company, the Team Lead needs to work on different branches and responsibilities. We don't think that the goal should be to develop all of the areas in the roadmap. It's important to focus on the problems and needs of your workplace, first of all.

Reviewer 2: ```suggestion
> 🚨 **Important:** Depending on a company, the Team Lead needs to work on different branches and responsibilities. We don't think that the goal should be to develop all of the areas in the roadmap. It's important to focus on the problems and needs of your workplace, first of all.
```

> The work is still in progress, so feel free to publish an Issue, drop your pull request and join the discussion [in Telegram chat, RU](https://tlinks.run/tlbootcamp).

Reviewer: Тут тоже про чатик спилить лучше

Reviewer 3: ```
open a your pull request or join a discussion
```

Author: fixed

# How to use it?
## 🏫 For a company

> 1. Download the [MindMap](roadmap.mm) with the full Team Lead model.

Reviewer 2: ```suggestion
1. Download the [MindMap](roadmap.mm) with the Team Lead model.
```

Author: Here, the "full" word has its special meaning, I don't think it should be removed.

Reviewer 2: It's not clear what is special in this meaning (do we have a short Team Lead model?) but I don't mind to leave it as-is.

Author: Yep, we have subsets of the model, in "examples"

Author: Let's see if there are some other opinions. As a variant, we can change "full" to "complete"

> 2. Go through all branches of the Team Lead responsibilities. Remove those that are not required by your company or are already being performed by someone else.

Reviewer 2: ```suggestion
2. Go through all the branches of the Team Lead responsibilities. Remove those that are not required by your company or are already being performed by someone else.
```
?

Author: Redundant, imho. That's definitely not a grammatical error and I don't see a reason to add more words here.

3. Form from the resulting map the artifacts you need: a profile to hire, a description of role expectations, a development plan.
4. Use our knowledge base to go deeper into any of the branches. For each of the branches we describe in detail its meaning, motivation to use, examples of good and bad behavior, ways of development in practice and theory.

## 👩🏼‍💻 For yourself

> 1. Download the [MindMap](roadmap.mm) with the full Team Lead model.

Reviewer: Нам надо либо сразу перевести майндмеп в этом PR, либо указать в описании, что он только на русском.

> 2. Mark on it the competencies you already have and those you need for further growth within the company. To give you a hint, look at what your manager or colleagues are doing. If there are still difficulties, ask a question [in our chat room, RU](https://tlinks.run/tlbootcamp).

Reviewer: Тоже чат

3. Make a list of competencies that that separate your current profile and your target profile.
4. Using our knowledge base, create a development plan for each of the competencies, which includes both theory and practical application.
5. Show your development plan to your manager and ask for assistance.

# Roadmap

![Team Lead Map](../roadmap.png)

The roadmap consists of two sections:

> - **Роли и обязанности.** A list of high-level work roles and more specific responsibilities.

Reviewer 2: ```suggestion
- **Roles and responsibilities.** A list of high-level work roles and more specific responsibilities.
```

Author: Good catch

- **Personal Skills.** Personal skills and qualities that are necessary for certain roles and responsibilities.

> Each of the map's leaves will eventually become a large entry in the knowledge base, containing a description, examples of behavior, and ways to develop the skill.

Reviewer 2: ```suggestion
Each of the map's leaves will eventually become a large entry in the knowledge base with a description, examples of behavior, and ways to develop the skill.
```
?

Author: Not sure. Other opinions?

Reviewer 3: @php-coder's suggestion makes more sense

## Download roadmap
- [PNG](../roadmap.png)
- [MindMap](../roadmap.mm)

> The easiest way to open a `mm` file is to visit [Freemind](https://sourceforge.net/projects/freemind/) (free and simple), [MindNode](https://mindnode.com/) (expensive and premium) or [MindMeister](https://mindmeister.com) (freemium and online). Other MindMap editors are also likely to work, but we haven't tested them.

Reviewer 2: ```suggestion
The easiest way to open a `mm` file is to use [Freemind](https://sourceforge.net/projects/freemind/) (free and simple), [MindNode](https://mindnode.com/) (expensive and premium) or [MindMeister](https://mindmeister.com) (freemium and online). Other MindMap editors are also likely to work, but we haven't tested them.
```

## Examples of use
We've prepared a few common Team Lead models. With their help, you can see how to work with the roadmap, or you can just bring them to your company as examples:
- Playing coach in a scrum team: [MindMap](../examples/playing-coach.mm)
- Technical Lead: [MindMap](../examples/techlead.mm)
- Startup Team Lead: [MindMap](../examples/startup-teamlead.mm)

# Work in progress

> The current roadmap is an alpha version that has a common teamwork structure and partially describes the first major role, Resource Manager. We are looking forward to your participation in the project, the process of contribution is described in detail in [`CONTRIBUTING.md`](CONTRIBUTING.md).

Reviewer 2: ```suggestion
The current roadmap is an alpha version that has a common teamwork structure and partially describes the first major role, Resource Manager. We are looking forward to your participation in the project. The contribution process is described in details in [`CONTRIBUTING.md`](CONTRIBUTING.md).
```
or
```suggestion
The current roadmap is an alpha version that has a common teamwork structure and partially describes the first major role, Resource Manager. We are looking forward to your participation in the project. The contribution process is described in depth in [`CONTRIBUTING.md`](CONTRIBUTING.md).
```

# The knowledge base
The base is gradually filled with Stas Tsyganov, Egor Tolstoy and the community. Our final goal is to describe each of the directions according to the following template:
- Description (what we are talking about)

> - Why is a branch important

Reviewer 2: ```suggestion
- Why this branch is important
```

- What happens if you don't work on it

> - Who can be delegated

Reviewer 2: ```suggestion
- To whom it can be delegated
```

    - Work on bugs
    - Incident management
    - Metrics and monitoring
    - Non-functional requirements
- Technical quality assurance
    - Unit tests
    - Pure code
    - Refactoring
    - Technical Debt Management
- Technology knowledge
    - Writing code
    - Technological stack
    - Selecting technologies and control of their stack
- Development cycle automation
    - VCS
    - CI
    - Release Automation
- Knowledge management
    - Knowledge sharing
    - Technical documentation
- Architecture
    - Technical requirements elicitation
    - Design
    - Architectural Reviews
    - Evolution
- Capacity Management

### Administrator
- Building the development cycle
    - Getting tasks
    - Development
        - Methodology design
            - Description
            - Evaluation
            - Execution
            - Control of execution
        - Ready-made approaches
            - Scrum
            - Lean
    - Issue of tasks
        - Acceptance
        - Rolling out
- Project management
    - Methodology design
    - Ready-made approaches
        - PMBoK
        - PDCA
- Stakeholder

## Personal Skills

#### Self development
- Ability to learn
- Reflexion

> - Making habits.

Reviewer 2: ```suggestion
- Making habits
```

> ![Logotype](../logo.png)

Reviewer 4: How about https://gitlocalize.com/ ? Keeping translation updated might be tricky

Author: Looks interesting. Can you Add the same comment to https://github.com/tlbootcamp/tlroadmap/pull/141, to leave this pr for one page translation and discuss the overall process in the other one?
